### PR TITLE
FIX: Allow non-native byte order inputs to scipy.fft

### LINKED ISSUE #10422: BUG: scipy.fft does not support multiprocessing
`scipy.fft` interacts badly with `joblib`, specifically when using the "processes" method. @peterbell10 tested and reports that it also fails for `multiprocessing.Pool` directly. But here was my MWE with joblib:
```Python
>>> import numpy as np
>>> from joblib import Parallel, delayed
>>> from scipy.fft import rfft
>>> [rfft(np.zeros(1)) for _ in range(2)]  # works
>>> Parallel(n_jobs=2, prefer='threads')(delayed(rfft)(np.zeros(1)) for _ in range(2))  # works
>>> Parallel(n_jobs=2, prefer='processes')(delayed(rfft)(np.zeros(1)) for _ in range(2))  # fails
joblib.externals.loky.process_executor._RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/home/larsoner/python/joblib/joblib/externals/loky/process_executor.py", line 418, in _process_worker
    r = call_item()
  File "/home/larsoner/python/joblib/joblib/externals/loky/process_executor.py", line 272, in __call__
    return self.fn(*self.args, **self.kwargs)
  File "/home/larsoner/python/joblib/joblib/_parallel_backends.py", line 567, in __call__
    return self.func(*args, **kwargs)
  File "/home/larsoner/python/joblib/joblib/parallel.py", line 225, in __call__
    for func, args, kwargs in self.items]
  File "/home/larsoner/python/joblib/joblib/parallel.py", line 225, in <listcomp>
    for func, args, kwargs in self.items]
  File "/home/larsoner/python/scipy/scipy/fft/_basic.py", line 298, in rfft
    return _pocketfft.rfft(x, n, axis, norm, overwrite_x)
  File "/home/larsoner/python/scipy/scipy/fft/_pocketfft/basic.py", line 134, in rfft
    return pfft.r2c(tmp, (axis,), True, norm, None, _default_workers)
RuntimeError: unsupported data type
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/larsoner/python/joblib/joblib/parallel.py", line 934, in __call__
    self.retrieve()
  File "/home/larsoner/python/joblib/joblib/parallel.py", line 833, in retrieve
    self._output.extend(job.get(timeout=self.timeout))
  File "/home/larsoner/python/joblib/joblib/_parallel_backends.py", line 521, in wrap_future_result
    return future.result(timeout=timeout)
  File "/usr/lib/python3.7/concurrent/futures/_base.py", line 432, in result
    return self.__get_result()
  File "/usr/lib/python3.7/concurrent/futures/_base.py", line 384, in __get_result
    raise self._exception
RuntimeError: unsupported data type
```
Modifying the SciPy code to print the dtype and `__class__` of at least `tmp` shows `np.float64` and `np.ndarray` as expected, so I'm not sure what unsupported datatype we are getting.

### PR DESCRIPTION
**Pull request by peterbell10**
#### Reference issue
Fixes https://github.com/scipy/scipy/issues/10422#issuecomment-587040183

#### What does this implement/fix?
`_pocketfft` transforms now accept inputs in non-native byte-order by copying the data to native byte-order first.

### CODE REVIEW

File: scipy/fft/_pocketfft/tests/test_basic.py [hunk 1]
```diff
@@ -1001,3 +1006,18 @@ def test_invalid_norm(func):
...
+@pytest.mark.parametrize('func', [fft, ifft, fftn, ifftn,
+                                   irfft, irfftn, hfft, hfftn])
+def test_swapped_byte_order_complex(func):
+    rng = np.random.RandomState(1234)
+    x = rng.rand(10) + 1j * rng.rand(10)
+    assert_allclose(func(swap_byteorder(x)), func(x))
+
+
+@pytest.mark.parametrize('func', [ihfft, ihfftn, rfft, rfftn])
+def test_swapped_byte_order_complex(func):
```
Comment (grlee77) [inline scipy/fft/_pocketfft/tests/test_basic.py]:
```suggestion
def test_swapped_byte_order_real(func):
```

**Review by grlee77 — CHANGES_REQUESTED** [review]
there is one duplicate test function name, but otherwise looks good to me

**Comment by grlee77** [comment]
There is also one error on Travis that appears to be a tolerance issue. 
https://travis-ci.org/scipy/scipy/jobs/651622854#L1379-L1390

**Comment by peterbell10** [comment]
Weird, the test failures are in `long double` which is only tested to the same precision as `double`.

**Review by grlee77 — APPROVED** [review]
looks like the failure was not reproduced after your commit renaming the test function

**Comment by tylerjereddy** [comment]
CI is all green, there's a +1 expert review, and it looks to me like there's some pretty thorough additional testing added here, so in it goes, thanks @peterbell10 & @grlee77

**Pull request merged**